src/core/opamConsole.ml

```diff
@@ -560,7 +560,10 @@ let clear_pending debug_level =
   Queue.iter f pending;
   Queue.clear pending
 
-let log section ?(level=1) fmt =
+let log_mutex = Mutex.create ()
+let log section ?(level=1) k =
+  Mutex.protect log_mutex @@ fun () ->
+  k @@ fun fmt ->
```

Comment:
this might not be necessary if we replace the use of the default formatters by their OCaml5 `Format.get_*` counterparts.